pyracing/client.py

```diff
@@ -534,74 +564,99 @@ async def results(
             'upperbound': upper_bound,
             'sort': sort,
             'order': order,
-            'format': format,
-            'category1': category1,
-            'category2': category2,
-            'category3': category3,
-            'category4': category4,
+            'format': data_format,
+            'category%5B%5D': category1,
+            'category%5B%5D': category2,
+            'category%5B%5D': category3,
+            'category%5B%5D': category4,
             'seasonyear': season_year,
             'seasonquarter': season_quarter,
             'raceweek': race_week,
             'trackid': track_id,
             'carclassid': car_class,
-            'carid': car_id
+            'carid': car_id,
+            'start_low': start_low,
+            'start_high': start_high,
+            'finish_low': finish_low,
+            'finish_high': finish_high,
+            'incidents_low': incidents_low,
+            'incidents_high': incidents_high,
+            'champpoints_low': champpoints_low,
+            'champpoints_high': champpoints_high
         }
         url = ct.URL_RESULTS
-        return await self.build_request(url, payload)
+        response = await self.build_request(url, payload)
+
+        # TODO Consider "try:" for a TypeError for response of {"m":{},"d":[]}
+        return [historical_data.EventResults(x) for x in response.json()["d"]["r"]]
 
-    async def season_for_session(self, sessionID):
+    async def season_from_session(self, sessionID):
         """ Returns the seasonID for a given sessionID. That is this endpoints
         only purpose.
         """
         payload = {'sessionID': sessionID}
         url = ct.URL_SEASON_FOR_SESSION
-        return await self.build_request(url, payload)
+        response = await self.build_request(url, payload)
+
+        return iracing_data.SeasonFromSession(response.content)
 
     async def season_standings(
             self,
             seasonID,
+            raceWeek=-1,
             carClassID=-1,
             clubID=-1,
-            raceWeek=-1,
             division=-1,
             start=1,
             end=25,
             sort=ct.Sort.champ_points,
             order=ct.Sort.descending
     ):
         """ Returns the championship point standings of a series.
-        This is the same data found in /statsseries.jsp
+        This is the same data found in /statsseries.jsp.
         """
-
         payload = {
             'seasonid': seasonID,
             'carclassid': carClassID,
             'clubid': clubID,
-            'raceweek': raceWeek,
+            'raceweek': raceWeek - 1,  # Makes human '1' == computer '1'
```

Comment:
What is the intention here?

Reply:
You assign it here then reassign it down below but I still dont understand what we want or what each number represents

Reply:
I've found that iRacing is not consistent for when list index format is followed. In this instance, before this change, entering a 1 in raceweek will return the results for week 2. subtracting 1 from the sent value re-aligns that when you see week 1 on a page from iRacing.com, you would expect to also receive week 1 when giving them raceweek=1 as a parameter.

Reply:
Oh, and so if you use -1, which is iRacing's value for 'return all,' then it would really submit -2. The conditional checks if you asked for -1, and if you did it actually sends -1.